### ISSUE TITLE
Incorrect verb "read" versus "get" in clusterscan resource RBAC prevents correctly granting clusterscan permissions

### ISSUE THREAD

**Issue by axeal** [rancher/rancher#30395]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create a Rancher v2.4.11 or v2.5.3 instance with a single all role custom cluster (I used https://github.com/superseb/tf-do-rancher2/).
2. Create a user `foo`
3. Create a custom Global role `CIS`, that grants `get` and `list` on `cisbenchmarkversions.management.cattle.io` and `cisconfigs.management.cattle.io`. Edit user `foo` and grant the custom `CIS` role.
4. Create a custom Cluster role `Clusterscans`, that grants `get` and `list` on `clusterscans.management.cattle.io`, and grant the user `foo` this role on the custom cluster.
5. As the admin user run a CIS Scan on the custom cluster.

**Result:**

- Logged in as the admin user observe once the CIS scan is complete that you can click the scan name to view the report details, and that viewing the scan in the API the `links: report:` field is populated.
- Logged in as the user `foo` observe that whilst the CIS Scan resource is listed, you cannot view the report details in the UI, and whilst you an access the `v3/clusterScans/<cluster_id>:<report_id>/report` API endpoint successfully, the `links: report` field in the `v3/clusterScans/<cluster_id>:<report_id>` endpoint is not populated (which is the reason the report details are not populated in the UI.

**Other details that may be helpful:**

The issue here is a result of the incorrect verb `read` versus the expected verb `get` in the API RBAC for the formatter that appends the report link to the clusterscan resource:

https://github.com/rancher/rancher/blob/v2.5.3/pkg/api/norman/customization/clusterscan/clusterscan.go#L28
https://github.com/rancher/rancher/blob/v2.4.11/pkg/api/customization/clusterscan/clusterscan.go#L26

Whilst an admin has `*` permission for all verbs, the defined custom role using the Kubernetes RBAC verb `get` does not match in the API check, hence the difference between the `links: report:` field returned for the admin user and the `foo` user. The verb in the API CanDo check should be updated to `get` to fix this issue. As a workaround the custom `Clusterscans` cluster role can be edited via the API to add the verb `read` to the list of permissions on `clusterscans.management.cattle.io`.

**Environment information**
- Rancher v2.4.11/v2.5.3

gz#13788

**Comment by doflamingo721** [comment]
@AndrewHoffmanQA Did you get a chance to reproduce this issue on rancher 2.6?

**Comment by AndrewHoffmanQA** [comment]
@doflamingo721 

I was able to reproduce this RBAC issue. I also found an additional issue (see additional context section). 


**Steps to reproduce** 
reproduced on a 2.6.6 HA server.

1. stand up rancher HA server (no downstream clusters needed) 
2. Create a user 
3. Create a custom Global role CIS, that grants get and list on cisbenchmarkversions.management.cattle.io and cisconfigs.management.cattle.io.

![image](https://user-images.githubusercontent.com/89030218/186194741-2c185f89-35c7-4027-87e6-d043b55f5dbe.png)

4.  Edit user foo and grant the custom CIS role.
5. Create a custom Cluster role Clusterscans, that grants get and list on clusterscans.management.cattle.io

![image](https://user-images.githubusercontent.com/89030218/186194662-6c24fb40-cc7d-490a-81c5-469114e43337.png)

6. grant the user foo this role on the custom cluster.
7. as admin, run a CIS benchmark scan with the  '1.6 permissive scan.' profile applied   
8. log in as the user you created
9. view CIS benchmark scan page
10. click on the name of the scan

**expected results** 

the scan displays a line item list of every pass/warning/failure.  (see in picture admin account can view line items fine.
![image](https://user-images.githubusercontent.com/89030218/186197388-42b0ee15-bafd-4142-b681-214ec786c27a.png)


**actual results** 

no scan data is displayed to the user.


**additional context** 

 when logged in as the user on the scan page, if you try to download the scan this error is displayed in the UI 

"Error downloading file
Cannot read properties of undefined (reading 'aggregatedTests')"

if they can view the report they should be able to download it, but if this restriction is by design we should replace the error message with whatever our standard RBAC error message is. ex: 'you do not have permissions to execute this action, please contact administrator' 

^^ after investigation if this needs its own separate ticket @AndrewHoffmanQA and i'll generate one

**Comment by rayandas** [comment]
I debugged and found the root cause of this issue. 
Which is: according to the snapshots above, we are giving permission only to resource `clusterscans` in cluster roles. But to view/download the cluster scan report we need to give permission to resource `clusterscanreports` as well, else the user `foo` can't view/download the report. 

**Solutions:**
There are two ways to fix this issue.
1. On UI select `clusterscanreports` resource in cluster role and give `get` and `list` verbs along with `clusterscans` resource. (`clusterscanreports` is not visible in the drop down list of resources currently. Figuring out if this is an UI issue or not. Then we can raise an UI ticket to solve this.)
2. Edit the cluster role manually using kubectl and add `clusterscanreports` in the resources. 
```
> kubectl edit clusterrole rt-hj477

apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    authz.cluster.cattle.io/clusterrole-owner: rt-hj477
  labels:
    cattle.io/creator: norman
  name: rt-hj477
rules:
- apiGroups:
  - cis.cattle.io
  resources:
  - clusterscans
  - clusterscanreports
  verbs:
  - get
  - list
```
Attaching the snapshot for reference.
![Screenshot from 2022-09-07 12-43-50](https://user-images.githubusercontent.com/32811240/188813324-79be01ce-58d0-4a72-b360-f4be304e4b80.png)

**Comment by prachidamle** [comment]
@rayandas Can we proceed to debug why the clusterscanreports resource is not visible in the drop down list of resources currently? Since this usecase is talking about creating a custom role on UI, if we figure out a way to display clusterscanreports as well that will be a good solution for this.

**Comment by rayandas** [comment]
I was debugging why the `clusterscanreports` resource is not visible on the UI.
Things are already written in rancher/cis-operator under `cis.cattle.io` api group. https://github.com/rancher/cis-operator/blob/master/pkg/generated/controllers/cis.cattle.io/v1/clusterscanreport.go. 
It's just not visible on the drop down list on the UI while creating cluster role. If we add `clusterscanreports` resource by editing `kubectl edit clusterrole <name>` then it’s working fine. I think it's an UI issue.

**Comment by gaktive** [comment]
@rayandas if the UI is the only work left for this ticket, then I can transfer this over to `rancher/dashboard`  for more immediate attention. Otherwise, I can make a new ticket based on the previous comment. What's the milestone for this?

**Comment by rayandas** [comment]
@gaktive Seems like there’s only UI work left. So either is fine. There’s no milestone for this issue as of now.

**Comment by gaktive** [comment]
Filed https://github.com/rancher/dashboard/issues/7035 for UI, though putting in 2.7.1 for now until triaged.

**Comment by rayandas** [comment]
Thanks @gaktive 
Moving this issue to blocked until the UI issue is fixed.

**Comment by gaktive** [comment]
@rayandas UI has a fix that's now in QA's hands so this should be unblocked now.

**Comment by rayandas** [comment]
Thanks @gaktive This is the last task for this issue. So I will move the the issue to review and then to test when the PR https://github.com/rancher/dashboard/pull/7259 will be merged. The milestone is 2.7.1 (UI issue) so I think the PR will be merged after 2.7.0 is release.

**Comment by AndrewHoffmanQA** [comment]
retest failed, passing this ticket back to dev. 

my understanding is that the bug has now shifted to a UI bug regarding the lack of `clusterscanreports ` resource being displayed in the UI when creating a RoleTemplate. per screenshot this option is still not available for users. 

tested on rancher 2.7head ce9a7ae

![image](https://user-images.githubusercontent.com/89030218/202518527-ea26af1a-49a9-45c1-8c90-e3e3a8d4067d.png)

**Comment by gaktive** [comment]
@AndrewHoffmanQA do we need to update https://github.com/rancher/dashboard/issues/7035 with the failure too? or is that being tested separately?

**Comment by AndrewHoffmanQA** [comment]
@gaktive you are definitely right, thanks for the callout. i'll kick that ticket back and keep this one in QA

**Comment by AndrewHoffmanQA** [comment]
tested in 2.7head f7692f3
retest successful. marking this ticket as done. 

Note: the bug was clusterscanreport not being available in the UI, this permission needs to be applied to the role for the user to see the scan.  following the repro steps above would result in the same failure and that is by design. marking as 'done' 



screenshot of user successfully viewing report inside UI as well as a downloaded copy. ![image.png](https://zube.io/files/rancher/600ba95befa77e59086cabff87606fba-image.png)
